distributed/scheduler.py

```diff
@@ -385,12 +387,81 @@ def mark_ready_to_run(self, key):
                 self.unrunnable.add(key)
             else:
                 self.stacks[new_worker].append(key)
-                self.maybe_ready.add(new_worker)
+                self.maybe_idle.add(new_worker)
                 # self.ensure_occupied(new_worker)
         else:
             self.ready.appendleft(key)
             # self.ensure_idle_ready()
 
+    def should_steal(self, key, bandwidth=100e6):
+        """ Is a key good for stealing from its chosen worker?
+
+        It must have the following attributes
+
+        1.  Not have too many dependencies
+        2.  Not be restricted to run on that worker
+        3   Take less time to transfer than to compute
+        """
+        if len(self.dependencies[key]) > 10:
```

Comment:
It could be, yes.  Generally dependencies tend to be in the 1-4 range or in the 1000s range, so comparing to ten is a fairly discriminating test.  It's also usually ok if we have false negatives.